cadence/contracts/connectors/SwapStack.cdc

```diff
@@ -292,15 +292,12 @@ access(all) contract SwapStack {
                 return // nothing to swap from, no capacity to ingest, invalid Vault type - do nothing
             }
 
-            let quote = self.swapper.quoteIn(forDesired: limit, reverse: false)
+            let quote = limit > from.balance
```

Comment:
The potential issue I see with this is that we're comparing amounts of two different currencies. `from.balance` is the amount of pre-conversion currency while `limit` defines the amount of post-conversion currency the inner `Sink` can accept.

So there's not necessarily a correlation between them, at least not enough to condition a quote on IMO